File: Exercises/Ex6/returned_answers/leonardo/makeTree.py
import ROOT
import random
from array import array

file = ROOT.TFile("random_numbers.root", "RECREATE")
tree = ROOT.TTree("tree", "Random Numbers")

x = array('d', [0])
tree.Branch("x", x, 'x/D')

for i in range(1000):
    x[0] = random.gauss(0,1)
    tree.Fill()
    
tree.Write()
print("Tree saved!")
file.Close()

File: Exercises/Ex6/returned_answers/leonardo/makeTree.cpp
#include <iostream>

#include <TRandom3.h>
#include <TFile.h>
#include <TTree.h>

int main(int argc, char** argv){
    TRandom3 rand;

    TFile* file = new TFile("random_numbers.root", "RECREATE");
    TTree* tree = new TTree("tree", "Random Numbers");

    double x;
    tree->Branch("x", &x, "x/D");

    for (int i = 0; i < 1000; i++){
        x = rand.Gaus(0, 1);
        tree->Fill();
    }

    tree->Write();
    file->Close();
    
    std::cout << "Tree saved!" << std::endl;
    return 0;
}
